platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -98,63 +106,50 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(HttpSolrClientFactory.class);
 
   @Override
-  public org.codice.solr.client.solrj.SolrClient newClient(String core) {
-    String solrUrl =
-        StringUtils.defaultIfBlank(
-            AccessController.doPrivileged(
-                (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
-            getDefaultHttpsAddress());
-    final String coreUrl = solrUrl + "/" + core;
-    final String solrDataDir =
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SOLR_DATA_DIR));
-
-    if (solrDataDir != null) {
-      ConfigurationStore.getInstance().setDataDirectoryPath(solrDataDir);
+  public org.codice.solr.client.solrj.SolrClient newClient(String coreName) {
+    Args.notEmpty(coreName, "Cannot create Solr client. Missing core name.");
+    if (getSolrDir() != null) {
+      ConfigurationStore.getInstance().setDataDirectoryPath(getSolrDir());
     }
-    LOGGER.debug("Solr({}): Creating an HTTP Solr client using url [{}]", core, coreUrl);
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+    LOGGER.debug(
+        "Solr({}): Creating an HTTP Solr client using url [{}]", coreName, getCoreUrl(coreName));
```

Comment:
❗️ The same was true for the url which is why it was originally stored into a local variable and passed down to the `createSolrHttpClient()` method.

Reply:
Think it will make much of a difference? How could I benchmark it to find out? I could write an integration or component test. The test would call the `newClient` 100 times. I could run the test for cached and uncached versions and compare the runtimes. If the different looks significant, I'll cache this and the other values.

Reply:
It is not a matter of benchmarking here. Going through the security manager or access controller should not be done in repetition. Values should be cached. There is a penalty involved each time you involve the security manager which is known to slow down the system.